src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -4188,8 +4188,7 @@ void ClassFileParser::set_precomputed_flags(InstanceKlass* ik) {
   // If it cannot be fast-path allocated, set a bit in the layout helper.
   // See documentation of InstanceKlass::can_be_fastpath_allocated().
   assert(ik->size_helper() > 0, "layout_helper is initialized");
-  if ((!RegisterFinalizersAtInit && ik->has_finalizer())
-      || ik->is_abstract() || ik->is_interface()
+  if (ik->is_abstract() || ik->is_interface()
```

Comment:
TIL what this bit was.  There are some comments in the interpreters that claim that the bit will be set if there are finalizers, which is no longer true.  The comments look like:

  // test to see if it has a finalizer or is malformed in some way

Reply:
Thanks.  I never would have found that!